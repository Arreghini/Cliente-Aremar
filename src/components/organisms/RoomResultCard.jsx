import React from 'react';

const RoomResultCard = ({ room, onReserve }) => {
  if (!room) return null;

  return (
    <div className="mt-6 flex flex-col items-center gap-4 border rounded-xl shadow-md p-4 bg-white">
      {room.photoRoom && (
        <img
          src={room.photoRoom}
          alt={`Foto de ${room.roomType?.name || 'habitación'}`}
          className="rounded-xl shadow-lg max-w-xs w-full h-auto"
        />
      )}
      <div  className="text-center">
        <h3 className="font-semibold text-lg">{room.roomType?.name}</h3>
        <p className="text-gray-600">Capacidad: {room.capacity} huéspedes</p>
        <p className="text-gray-600">Precio: ${room.price}/noche</p>
      </div>
      <button
        onClick={onReserve}
        className="px-4 py-2 bg-green-500 text-white rounded-md hover:bg-green-600 transition"
      >
        Reservar ahora
      </button>
    </div>
  );
};

export default RoomResultCard;
