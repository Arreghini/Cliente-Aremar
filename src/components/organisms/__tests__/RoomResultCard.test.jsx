import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import RoomResultCard from '../RoomResultCard';

describe('RoomResultCard component', () => {
  const mockRoom = {
    photoRoom: 'https://example.com/photo.jpg',
    roomType: { name: 'Habitación Deluxe' },
    capacity: 3,
    price: 150,
  };

  it('1. No renderiza nada si no recibe room', () => {
    const { container } = render(<RoomResultCard room={null} onReserve={vi.fn()} />);
    expect(container.firstChild).toBeNull();
  });

  it('2. Renderiza los datos de la habitación correctamente', () => {
    render(<RoomResultCard room={mockRoom} onReserve={vi.fn()} />);
    expect(screen.getByText('Habitación Deluxe')).toBeInTheDocument();
    expect(screen.getByText(/Capacidad: 3 huéspedes/i)).toBeInTheDocument();
    expect(screen.getByText(/Precio: \$150\/noche/i)).toBeInTheDocument();
  });

  it('3. Muestra la imagen si photoRoom existe', () => {
    render(<RoomResultCard room={mockRoom} onReserve={vi.fn()} />);
    const img = screen.getByAltText('Foto de Habitación Deluxe');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', mockRoom.photoRoom);
  });

  it('4. No muestra imagen si photoRoom no existe', () => {
    const roomWithoutPhoto = { ...mockRoom, photoRoom: null };
    render(<RoomResultCard room={roomWithoutPhoto} onReserve={vi.fn()} />);
    expect(screen.queryByRole('img')).not.toBeInTheDocument();
  });

  it('5. Ejecuta onReserve al hacer click en el botón', () => {
    const onReserveMock = vi.fn();
    render(<RoomResultCard room={mockRoom} onReserve={onReserveMock} />);
    const button = screen.getByRole('button', { name: /reservar ahora/i });
    fireEvent.click(button);
    expect(onReserveMock).toHaveBeenCalledTimes(1);
  });

  it('6. Match con snapshot (estructura básica)', () => {
    const { container } = render(<RoomResultCard room={mockRoom} onReserve={vi.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
