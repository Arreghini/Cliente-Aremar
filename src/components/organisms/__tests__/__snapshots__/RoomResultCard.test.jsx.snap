// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoomResultCard component > 6. Match con snapshot (estructura básica) 1`] = `
<div>
  <div
    class="mt-6 flex flex-col items-center gap-4 border rounded-xl shadow-md p-4 bg-white"
  >
    <img
      alt="Foto de Habitación Deluxe"
      class="rounded-xl shadow-lg max-w-xs w-full h-auto"
      src="https://example.com/photo.jpg"
    />
    <div
      class="text-center"
    >
      <h3
        class="font-semibold text-lg"
      >
        Habitación Deluxe
      </h3>
      <p
        class="text-gray-600"
      >
        Capacidad: 
        3
         huéspedes
      </p>
      <p
        class="text-gray-600"
      >
        Precio: $
        150
        /noche
      </p>
    </div>
    <button
      class="px-4 py-2 bg-green-500 text-white rounded-md hover:bg-green-600 transition"
    >
      Reservar ahora
    </button>
  </div>
</div>
`;
